src/http/io.rs

```diff
@@ -102,13 +97,15 @@ impl<T: AsyncRead + AsyncWrite> Buffered<T> {
                 Ok(n) => n,
                 Err(e) => {
                     if e.kind() == io::ErrorKind::WouldBlock {
+                        // TODO: Push this out, ideally, into http::Conn.
                         self.read_blocked = true;
```

Comment:
Returning an `Err` from `Stream::poll` signals the stream is doomed. Tokio will drop the stream.